src/plugins/elasticsearch/lib/create_proxy.js

```diff
@@ -12,7 +12,8 @@ function createProxy(server, method, route, config) {
         mapUri: mapUri(server),
         passThrough: true,
         agent: createAgent(server),
-        xforward: true
+        xforward: true,
+        timeout: server.config().get('elasticsearch.requestTimeout') + 100
```

Comment:
I had put this in to prevent a race condition and allow the web es client (https://github.com/ich199/kibana/blob/4.x/src/ui/public/es.js) to return/display the correct requestTimeout error before the socket was closed by the Kibana server process. Whether it is actually needed, I'm not sure.